src/Surfnet/StepupSelfService/SelfServiceBundle/Tests/Service/RemoteVetting/RemoteVettingContextTest.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupSelfService\SelfServiceBundle\Tests\Service\RemoteVetting;
+
+use PHPUnit_Framework_TestCase as UnitTest;
+use Surfnet\StepupMiddlewareClientBundle\Uuid\Uuid;
+use Surfnet\StepupSelfService\SelfServiceBundle\Exception\InvalidRemoteVettingContextException;
+use Surfnet\StepupSelfService\SelfServiceBundle\Exception\InvalidRemoteVettingStateException;
+use Surfnet\StepupSelfService\SelfServiceBundle\Service\RemoteVetting\Dto\RemoteVettingTokenDto;
+use Surfnet\StepupSelfService\SelfServiceBundle\Service\RemoteVetting\RemoteVettingContext;
+use Surfnet\StepupSelfService\SelfServiceBundle\Service\RemoteVetting\ValueObject\ProcessId;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpFoundation\Session\Storage\MockArraySessionStorage;
+
+class RemoteVettingContextTest extends UnitTest
```

Comment:
Some remarks about this test

1. Idea, instead of importing the `PHPUnit_Framework_TestCase` aliased `UnitTest`, consider naming it `IntegrationTest`?
2. Not all `InvalidRemoteVettingStateException` possibilities are covered in test here. Not covered are: `handleInitialise`, `handleValidating`, and `handleValidated`. 
3. Consider not only expecting the exceptions being thrown in the 'sad flow' tests (`$this->expectException(InvalidRemoteVettingContextException::class);`). But consider also testing for the actual validation exception message. That way it becomes more clear what error state you are testing.

Reply:
1. :+1: 
1. :+1: Added additional tests
1. :+1: Added validation on messages to make things a bit more clear

Reply:
I also added a end-to-end (acceptance) test for remote vetting. This will also test the `SamlCalloutHelper`.

Reply:
2. Not all possible exceptions have been tested yet. Did you verify if these are impossible states? By running the test with code coverage, you can more easily see what im rambling on about.

Reply:
2. Added  a few tests more but it's not possible to get code coverage 100% because how it's implemented currently a process could always be started. And a to get to the next step a process should be started which will already be handled by the RemoteVettingContext.

Reply:
Superb! Thanks